examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython2",
+   "version": "2.7.3"
+  },
+  "name": ""
+ },
+ "nbformat": 3,
+ "nbformat_minor": 0,
+ "worksheets": [
+  {
+   "cells": [
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "# Associative Memory\n",
+      "\n",
+      "This tutorial introduces the Associative Memory (AM) module in the SPA.\n",
+      "\n",
+      "Associative memory is a neural network used to store and recall patterns.\n",
+      "When the network receives a partial or noisy pattern at the input, it can either recover the same pattern or recall another stored pattern. \n",
+      "If the recovered pattern is same as the input pattern, the memory is said to be autoassociative or \"clean-up\" memory. Otherwise, if the recovered pattern is different from the presented one, the network is heteroassociative. \n",
+      "\n",
+      "Patterns stored by the AM module in the SPA are semantic pointers organised in a SPA vocabulary.\n",
+      "The examples in this tutorial demonstrate how to use the AM module, store and recall patterns.\n",
+      "Advanced functionality of the module such as the recall of multiple memories similar to the input are also presented."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "# python libraries and plotting tools\n",
+      "import numpy as np\n",
+      "import matplotlib.pyplot as plt\n",
+      "%matplotlib inline\n",
+      "\n",
+      "import nengo\n",
+      "from nengo import spa\n",
+      "%load_ext nengo.ipynb"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "## Step 1: Creating an Associative Memory\n",
+      "\n",
+      "We start by creating a set of patterns the AM will store. \n",
+      "The vocabulary in this example contains five words: `ORANGE, APRICOT, CHERRY, STRAWBERRY` and `APPLE`.\n",
+      "Each word is represented as a semantic pointer, a $n$-dimensional vector.\n",
+      "When creating a vocabulary, we specify the number of dimensions for all semantic pointers.\n",
+      "Then, we add the words to a vocabulary with `parse`:"
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": true,
+     "input": [
+      "dim = 32\n",
+      "vocab = spa.Vocabulary(dimensions=dim)\n",
+      "\n",
+      "words = ['ORANGE', 'APRICOT', 'CHERRY', 'STRAWBERRY', 'APPLE']\n",
+      "\n",
+      "for word in words:\n",
+      "    vocab.parse(word)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "Next, we create an autoassociative memory with a corresponding set of stored patterns.\n",
+      "The memory is created within a `nengo.Network` module."
```

Comment:
Wait, the PR #797 is using associative memory just in tests, or am I not
seeing non-test related examples?

On 13 November 2015 at 19:37, Jan Gosmann <notifications@github.com> wrote:

> In examples/spa/associative_memory.ipynb
> <https://github.com/nengo/nengo/pull/895#discussion_r44849900>:
>
> > +      "\n",
> > +      "words = ['ORANGE', 'APRICOT', 'CHERRY', 'STRAWBERRY', 'APPLE']\n",
> > +      "\n",
> > +      "for word in words:\n",
> > +      "    vocab.parse(word)"
> > +     ],
> > +     "language": "python",
> > +     "metadata": {},
> > +     "outputs": []
> > +    },
> > +    {
> > +     "cell_type": "markdown",
> > +     "metadata": {},
> > +     "source": [
> > +      "Next, we create an autoassociative memory with a corresponding set of stored patterns.\n",
> > +      "The memory is created within a `nengo.Network` module."
>
> Note, the PR #797 <https://github.com/nengo/nengo/pull/797> introduces a
> basic associative memory implementation that is a normal network (not part
> of SPA) and a SPA implementation based on this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nengo/nengo/pull/895/files#r44849900>.
>